### PR TITLE
Added Focal tversky loss

### PR DESCRIPTION
**Pull request by Aditi-Mhatre**
Hi, 

Added: 
- Focal Tversky loss function in Focal_Tversky.py based on the [paper](https://arxiv.org/abs/1810.07842) 
- tests for the loss function (all passed so far)

Changed:
- losses -> _init_ and Focal_Tversky
- tests -> test_losses

Maybe you can take a look at it.

### CODE REVIEW

File: segmentation_models_pytorch/losses/focal_tversky.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        log_loss: bool = False,
+        from_logits: bool = True,
+        smooth: float = 0.0,
+        ignore_index: Optional[int] = None,
+        eps: float = 1e-7,
+        alpha: float = 0.5,
+        beta: float = 0.5,
+        gamma: float = 1.0,
+    ):
+        assert mode in {BINARY_MODE, MULTILABEL_MODE, MULTICLASS_MODE}
```
Comment (qubvel) [inline segmentation_models_pytorch/losses/focal_tversky.py:52]: Please move the class docstring under __init__ similar to other losses

File: segmentation_models_pytorch/losses/_functional.py [hunk 2]
```diff
@@ -205,6 +205,37 @@ def soft_tversky_score(
...
+    target_sum = torch.sum(target, dim=dims)
+    difference = LA.vector_norm(output - target, ord=1, dim=dims)
+
+    intersection = (output_sum + target_sum - difference) / 2  # TP
+    fp = output_sum - intersection
+    fn = target_sum - intersection
+
+    tversky_score = (intersection + smooth) / (
+        intersection + alpha * fp + beta * fn + smooth
+    ).clamp_min(eps)
```
Comment (qubvel) [inline segmentation_models_pytorch/losses/_functional.py:236]: As far as I understand we can just call `soft_tversky_score` here?

File: tests/test_losses.py [hunk 3]
```diff
@@ -92,6 +93,23 @@ def test_soft_tversky_score(y_true, y_pred, expected, eps, alpha, beta):
     actual = F.soft_tversky_score(y_pred, y_true, eps=eps, alpha=alpha, beta=beta)
     assert float(actual) == pytest.approx(expected, eps)
 
+@pytest.mark.parametrize(
+    ["y_true", "y_pred", "expected", "eps", "alpha", "beta", "gamma"],
+    [
+        [[1, 1, 1, 1], [1, 1, 1, 1], 0.0, 1e-5, 0.5, 0.5, 1.0], 
+        [[0, 1, 1, 0], [0, 1, 1, 0], 0.0, 1e-5, 0.5, 0.5, 1.0], 
+        [[1, 1, 1, 1], [0, 0, 0, 0], 1.0, 1e-5, 0.5, 0.5, 1.0], 
+        [[1, 0, 1, 0], [0, 1, 0, 0], 1.0, 1e-5, 0.5, 0.5, 1.0],
```
Comment (qubvel) [inline tests/test_losses.py:102]: Can you please add a test case with gamma != 1?

**Review by qubvel — COMMENTED** [review]
Hi @Aditi-Mhatre, thanks for adding a new loss function and thank you for adding a test!

I left some comments below, and can you please include the loss to the `docs/`? Thanks 🤗

**Comment by zifuwanggg** [comment]
The focal Tversky loss has already been already [implemented](https://github.com/qubvel-org/segmentation_models.pytorch/blob/0e2da9e3cbbae95512496147a034e525464a7b69/segmentation_models_pytorch/losses/tversky.py#L57).

**Comment by qubvel** [comment]
Indeed 😄 missed it, probably a documentation issue! Thanks for pointing it @zifuwanggg

File: tests/test_losses.py (hunk 3 above)
Comment (Aditi-Mhatre) [reply tests/test_losses.py:102]: Hi @qubvel, thanks a lot for your comments ! Since @zifuwanggg mentioned that the loss function is already implemented I'll close this pull request
